docs/code-style-guide.md

```diff
@@ -390,3 +390,22 @@ Some general guidelines when writing tests are:
 ### Protobuf
 
 Protobufs that we define should follow [Google's Protobuf Style Guide](https://developers.google.com/protocol-buffers/docs/style).
+
+* When initializing a protobuf from a unique pointer, avoid passing the `release()`'d pointer to `set_allocated` and `AddAllocated`. Releasing unique pointers makes it too easy to cause memory leaks.
+  ```cpp
+  /** Context:
+   * ball_state is field of vision_msg
+   * segments is google::protobuf::RepeatedPtrField<SSL_FieldLineSegment>
+   * tbots_robot_msgs is a repeated field in sensor_msg_1
+   */
+
+  // Incorrect
+  vision_msg->set_allocated_ball_state(createBallStateMsg(world.ball()).release());
```

Comment:
Would be good to make this example code more standalone, currently it's just a snippet from real code meaning most people don't know the return type of `createBallStateMsg` for example, or `world.ball()`

Reply:
I've added a snippet of the proto definition to make it more clear